### PR TITLE
Add M4 processor

### PR DESCRIPTION
**Pull request by kiracus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

add m4 cpu value

### CODE REVIEW

File: Library/Homebrew/test/hardware/cpu_spec.rb [hunk 1]
```diff
@@ -36,6 +36,7 @@
         :arm_lobos,
         :arm_monsoon_mistral,
         :arm_palma,
+        :arm_donan,
```
Comment (p-linnane) [inline Library/Homebrew/test/hardware/cpu_spec.rb:39]: Please alphabetize this in the list.

**Review by p-linnane — APPROVED** [review]

File: Library/Homebrew/test/hardware/cpu_spec.rb (hunk 1 above)
Comment (kiracus) [reply Library/Homebrew/test/hardware/cpu_spec.rb]: thanks. fixed

**Review by MikeMcQuaid — APPROVED** [review]
Thanks @kiracus!

**Comment by MikeMcQuaid** [comment]
Thanks so much for your first contribution! Without people like you submitting PRs we couldn't run this project. You rock, @kiracus!

**Pull request merged**